File: FUWEB/PHP/app/Http/Controllers/assetController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;
use Illuminate\Support\Facades\DB;
use Illuminate\Support\Facades\Log;
use App\Http\Controllers\Controller;

class assetController extends Controller
{
    /**
     * Get an asset by its ID.
     *
     * @param Request $request
     * @return \Illuminate\Http\JsonResponse
     */
    public function getAssetById(Request $request)
    {
        $assetId = $request->route('id');

        $inputData = ['id' => $assetId];

        try {
            $dbOutput = DB::select('CALL sp_getAsset(?)', [json_encode($inputData)]);
            $result = json_decode($dbOutput[0]->result, true);
            $statusCode = $result['status_code'];
            $message = $result['message'];
            if (isset($result['data'])) {
                $data = $result['data'];
                return response()->json([
                    'message' => $message,
                    'data' => $data,
                ], $statusCode);
            } else {
                return response()->json([
                    'message' => $message,
                ], $statusCode);
            }
        } catch (\Exception $error) {
            \Log::error($error);
            return response()->json(['error' => 'Internal Server Error'], 500);
        }
    }
    
    /**
     * Get all assets.
     *
     * @param Request $request
     * @return \Illuminate\Http\JsonResponse
     */
    public function getAllAssets(Request $request)
    {
        try {
            $dbOutput = DB::select('CALL sp_getAssets()');
            $result = json_decode($dbOutput[0]->result, true);
            $statusCode = $result['status_code'];
            $message = $result['message'];
            $data = $result['data'];

            return response()->json([
                'message' => $message,
                'data' => $data,
            ], $statusCode);
        } catch (\Exception $error) {
            \Log::error($error);
            return response()->json(['error' => 'Internal Server Error'], 500);
        }
    }

    /**
     * Get assets owned by a specific player.
     *
     * @param Request $request
     * @return \Illuminate\Http\JsonResponse
     */
    public function getPlayerAssets(Request $request)
    {
        $id = $request->route('id');
        $inputData = ['id' => $id];

        try {
            $dbOutput = DB::select('CALL sp_getUserAssets(?)', [json_encode($inputData)]);
            $result = json_decode($dbOutput[0]->result, true);
            $statusCode = $result['status_code'];
            $message = $result['message'];
            if (isset($result['data'])) {
                $data = $result['data'];
                return response()->json([
                    'message' => $message,
                    'data' => $data,
                ], $statusCode);
            } else {
                return response()->json([
                    'message' => $message,
                ], $statusCode);
            }
        } catch (\Exception $error) {
            \Log::error($error);
            return response()->json(['error' => 'Internal Server Error'], 500);
        }
    }

    /**
     * Buy an asset for a specific player.
     *
     * @param Request $request
     * @return \Illuminate\Http\JsonResponse
     */
    public function buyAsset(Request $request){
        try{
            $userId = $request->input('userId');
            $assetId = $request->input('assetId');
            $inputData = ['userId' => $userId,'assetId',$assetId];
            $dbOutput = DB::select('sp_buyAsset(?)', [json_encode($inputData)]);
            $result = json_decode($dbOutput[0]->result, true);
            $statusCode = $result['status_code'];
            $message = $result['message'];
            if (isset($result['asset'])) {
                $asset = $result['asset'];
                return response()->json([
                    'message' => $message,
                    'asset' => $asset,
                ], $statusCode);
            }else {
                return response()->json([
                    'message' => $message,
                ], $statusCode);
            }

        } catch (error) {
            \Log::error($error);
            return response()->json(['error' => 'Internal Server Error'], 500);
        }
    }

    /**
     * Get assets at a specific station.
     *
     * @param Request $request
     * @return \Illuminate\Http\JsonResponse
     */
    public function getAssetsStation(Request $request){
        try{
            $assetId = $request->input('assetId');
            $inputData = ['assetId' => $assetId];
            $dbOutput = DB::select('sp_getAssetsStation(?)', [json_encode($inputData)]);
            $result = json_decode($dbOutput[0]->result, true);
            $statusCode = $result['status_code'];
            $message = $result['message'];
            if (isset($result['data'])) {
                $data = $result['data'];
                return response()->json([
                    'message' => $message,
                    'data' => $data,
                ], $statusCode);
            }else {
                return response()->json([
                    'message' => $message,
                ], $statusCode);
            }

        } catch (error) {
            \Log::error($error);
            return response()->json(['error' => 'Internal Server Error'], 500);
        }
    }
    
    /**
     * Get assets in a specific world.
     *
     * @param Request $request
     * @return \Illuminate\Http\JsonResponse
     */
    public function getWorldAssets(Request $request){
        $worldId = $request->route('id');
        $inputData = ['worldId' => $worldId];
        try{
            $dbOutput = DB::select('sp_getWorldAssets(?)', [json_encode($inputData)]);
            $result = json_decode($dbOutput[0]->result, true);
            $statusCode = $result['status_code'];
            $message = $result['message'];
            if (isset($result['data'])) {
                $data = $result['data'];
                return response()->json([
                    'message' => $message,
                    'data' => $data,
                ], $statusCode);
            }else {
                return response()->json([
                    'message' => $message,
                ], $statusCode);
            }

        } catch (error) {
            \Log::error($error);
            return response()->json(['error' => 'Internal Server Error'], 500);
        }
    }
}

File: JAVA/railway/src/main/java/api/railway/Controller/AssetController.java
package api.railway.Controller;

import java.util.Map;

import org.json.JSONArray;
import org.json.JSONObject;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;
import api.railway.Database.AssetDAO;

@RestController
public class AssetController {

  private AssetDAO assetDAO = new AssetDAO();

  @GetMapping("/assets")
  public ResponseEntity<Object> getAssets() {
    JSONObject result = assetDAO.getAssets();

    int status_code = result.getInt("status_code");
    String message = result.getString("message");
    JSONArray data = result.getJSONArray("data");

    JSONObject responseJson = new JSONObject();
    responseJson.put("data", data);
    responseJson.put("message", message);

    return ResponseEntity.status(status_code).body(responseJson.toMap());
  }

  @GetMapping("/asset/{id}")
  public ResponseEntity<Object> getAsset(@PathVariable Long id) {
    JSONObject inputJSON = new JSONObject();
    inputJSON.put("id", id);
    JSONObject result = assetDAO.getAssetById(inputJSON);
    int status_code = result.getInt("status_code");
    String message = result.getString("message");
    JSONObject data = result.optJSONObject("data");

    JSONObject responseJson = new JSONObject();
    if (data != null)
      responseJson.put("data", data);
    responseJson.put("message", message);
    return ResponseEntity.status(status_code).body(responseJson.toMap());
  }

  @GetMapping("/asset/player/{id}")
  public ResponseEntity<Object> getPlayerAssets(@PathVariable Long id) {
    JSONObject inputJSON = new JSONObject();
    inputJSON.put("id", id);
    JSONObject result = assetDAO.getUserAssets(inputJSON);
    int status_code = result.getInt("status_code");
    String message = result.getString("message");
    JSONObject data = result.optJSONObject("data");

    JSONObject responseJson = new JSONObject();
    if (data != null)
      responseJson.put("data", data);
    responseJson.put("message", message);
    return ResponseEntity.status(status_code).body(responseJson.toMap());
  }

  @PostMapping("/asset/buy")
  public ResponseEntity<Object> buyAsset(@RequestBody Map<String, Object> requestBody) {
    String userId = (String) requestBody.get("userId");
    int assetId = (int) requestBody.get("assetId");
    JSONObject inputJSON = new JSONObject();
    inputJSON.put("userId", userId);
    inputJSON.put("assetId", assetId);
    JSONObject result = assetDAO.buyAsset(inputJSON);
    int status_code = result.getInt("status_code");
    String message = result.getString("message");
    JSONObject data = result.optJSONObject("asset");

    JSONObject responseJson = new JSONObject();
    if (data != null) {
      responseJson.put("data", data);
    }
    responseJson.put("message", message);
    return ResponseEntity.status(status_code).body(responseJson.toMap());
  }

  @PostMapping("/asset/station")
  public ResponseEntity<Object> getAssetsStation(@RequestBody Map<String, Object> requestBody) {
    int assetId = (int) requestBody.get("assetId");
    JSONObject inputJSON = new JSONObject();
    inputJSON.put("assetId", assetId);
    JSONObject result = assetDAO.getAssetsStation(inputJSON);
    int status_code = result.getInt("status_code");
    String message = result.getString("message");
    JSONObject data = result.optJSONObject("data");

    JSONObject responseJson = new JSONObject();
    if (data != null) {
      responseJson.put("data", data);
    }
    responseJson.put("message", message);
    return ResponseEntity.status(status_code).body(responseJson.toMap());
  }

  @GetMapping("/asset/world/{id}")
  public ResponseEntity<Object> getWorldAssets(@PathVariable Long id) {
    JSONObject inputJSON = new JSONObject();
    inputJSON.put("worldId", id);
    JSONObject result = assetDAO.getUserAssets(inputJSON);
    int status_code = result.getInt("status_code");
    String message = result.getString("message");
    JSONObject data = result.optJSONObject("data");

    JSONObject responseJson = new JSONObject();
    if (data != null)
      responseJson.put("data", data);
    responseJson.put("message", message);
    return ResponseEntity.status(status_code).body(responseJson.toMap());
  }

}
